todo_app/data/trello_items.py

```diff
@@ -0,0 +1,198 @@
+        "GET",
+        url,
+        headers=headers,
+        params=query)
+
+    names = []
+
+    if response.status_code == 200:
+        lists = json.loads(response.text)
+        for l in lists:
+            names.append(l['name'])
+            print(f"List id{l['id']} and name {l['name']}")
+    
+    return names 
+
+def _get_list_id(list_name):
+
+    url = f"https://api.trello.com/1/boards/{trello_board}/lists"
+
+    response = requests.request(
+        "GET",
+        url,
+        headers=headers,
+        params=query)
+
+    todo_list_id = ""
+
+    if response.status_code == 200:
+        lists = json.loads(response.text)
+
+        for l in lists:
+            if l['name'] == list_name:
+                todo_list_id = l['id']
+    else:
+        print(f"_get_list_id request has failed. Status code: {response.status_code}")
+
+    return todo_list_id
+class Item:
+    def __init__(self, id, name, status = 'To Do'):
+        self.id = id
+        self.name = name
+        self.status = status
+
+    @classmethod
+    def from_trello_card(cls, card, list):
+        return cls(card['id'], card['name'], list['name'])
+
+def _get_cards_in_list(list_id, list_name):
+    
+    url = f"https://api.trello.com/1/lists/{list_id}/cards"
+
+    response = requests.request(
+        "GET",
+        url,
+        headers=headers,
+        params=query)
+
+    items = []
+    
+    if response.status_code == 200:
+        cards = json.loads(response.text)
+
+        for c in cards:
+            i = Item(c['idShort'], c['name'], list_name)
+            items.append(i)        
+    else:
+        print(f"_get_cards_in_list request has failed. Status code: {response.status_code}")
+    
+    return items
+
+def _get_card_id(list_id, id_short : int):
+    
+    url = f"https://api.trello.com/1/lists/{list_id}/cards"
+
+    response = requests.request(
+        "GET",
+        url,
+        headers=headers,
+        params=query)
+
+    if response.status_code == 200:
+        cards = json.loads(response.text)
+       
+        for c in cards:
+            if c['idShort'] == id_short:
+                return c['id']
+    else:
+        print(f"_get_card_id request has failed. Status code: {response.status_code}")
+    
+    return ""
+
+def _add_card(list_id, card_name):
+
+    url = "https://api.trello.com/1/cards"
+
+    query['name'] = card_name
+    query['idList'] = list_id
+
+    response = requests.request(
+        "POST",
+        url,
+        headers=headers,
+        params=query
+    )
+
+    print(response.request.url)
+    print(response.request.body)
+    print(response.request.headers)
+
+def get_items():
+    """
+    Fetches items from the Trello API.
+
+    Returns:
+        list: The list of saved items.
+    """
+    all_items = []
+    lists_names = _get_lists()
+
+    for n in lists_names:
+        cards = _get_cards_in_list(_get_list_id(n),n)
+        for c in cards:
+            all_items.append(c)
```

Comment:
Not a problem, but for me this runs _fairly_ slowly - the main reason being that it's making a new HTTP request for every single list. If we instead used the `cards = open` parameter in your call to the `/lists` endpoint, then we could compress this down to just one request - which would also involve switching this loop to be more like:
```
for list in lists:
    cards = n['cards']
    for card in cards:
        all_items.append(Item.from_trello(card, list)
```